src/javascript/app_2/Stores/Modules/Portfolio/portfolio_store.js

```diff
@@ -0,0 +1,144 @@
+
+        WS.portfolio().then((response) => {
+            this.is_loading = false;
+            this.updatePortfolio(response);
+        });
+        WS.subscribeTransaction(this.transactionHandler, false);
+    };
+
+    @action.bound
+    clearTable() {
+        this.data       = [];
+        this.is_loading = false;
+        this.error      = '';
+    }
+
+    @action.bound
+    transactionHandler(response) {
+        if ('error' in response) {
+            this.error = response.error.message;
+        }
+        WS.portfolio().then((res) => this.updatePortfolio(res));
+    };
+
+    @action.bound
+    proposalOpenContractHandler(response) {
+        if ('error' in response) {
+            return;
+        }
+        const proposal = response.proposal_open_contract;
+
+        // force to sell the expired contract, in order to remove from portfolio
+        if (+proposal.is_settleable === 1 && !proposal.is_sold) { WS.sellExpired(); }
+
+        const position_data_index = this.data.findIndex(
+            (position) => position.id === +proposal.contract_id
+        );
+
+        if (position_data_index === -1) return;
+
+        if (proposal.is_sold) {
+            this.data.splice(position_data_index, 1);
+        } else {
+            const portfolio_position = this.data[position_data_index];
+
+            const prev_indicative = portfolio_position.indicative;
+            const new_indicative  = +proposal.bid_price;
+
+            portfolio_position.indicative = new_indicative;
+
+            if (!proposal.is_valid_to_sell) {
+                portfolio_position.status = 'no-resale';
+            }
+            else if (new_indicative > prev_indicative) {
+                portfolio_position.status = 'price-moved-up';
+            }
+            else if (new_indicative < prev_indicative) {
+                portfolio_position.status = 'price-moved-down';
+            }
+            else {
+                portfolio_position.status = 'price-stable';
+            }
+        }
+    }
+
+    @action.bound
+    updatePortfolio(response) {
+        if ('error' in response) {
+            this.error = response.error.message;
+            return;
+        }
+        if (response.portfolio.contracts && response.portfolio.contracts.length !== 0) {
+            this.data = formatPortfolioResponse(response.portfolio.contracts);
+            WS.subscribeProposalOpenContract(this.proposalOpenContractHandler, false);
+        }
+    }
+
+    @action.bound
+    onMount() {
+        this.initializePortfolio();
+    }
+
+    @action.bound
+    onUnmount() {
+        this.clearTable();
+        WS.forgetAll('proposal_open_contract', 'transaction');
+    }
+
+    @computed
+    get data_with_remaining_time() {
+        return this.data.map((portfolio_pos) => {
+            portfolio_pos.remaining_time = formatDuration(getDiffDuration(this.common.server_time, portfolio_pos.expiry_time));
+            return portfolio_pos;
+        });
```

Comment:
don't use es6 spread here, since it runs every second
this approach is 20 times faster than with es6 spread (`...portfolio_pos`)
[performance benchmark](http://jsben.ch/YTUEK)

Reply:
@binary-hamedanchi @morteza-binary should i add this to comments? 🤔

Reply:
@ruskakimov 
IMO, it's better to add it to comments, so it prevents others to change it unwittingly.